test/feature_library/test_feature_library.py

```diff
@@ -196,10 +483,527 @@ def test_concat():
         FourierLibrary(),
         PolynomialLibrary() + FourierLibrary(),
         pytest.lazy_fixture("data_custom_library"),
+        pytest.lazy_fixture("data_spatiotemporal_library"),
+        pytest.lazy_fixture("data_ode_library"),
+        pytest.lazy_fixture("data_pde_library"),
+        pytest.lazy_fixture("data_sindypi_library"),
     ],
 )
 def test_not_fitted(data_lorenz, library):
     x, t = data_lorenz
 
     with pytest.raises(NotFittedError):
         library.transform(x)
+
+
+@pytest.mark.parametrize(
+    "library",
+    [
+        IdentityLibrary(),
+        PolynomialLibrary(),
+        FourierLibrary(),
+        PolynomialLibrary() + FourierLibrary(),
+    ],
+)
+def test_library_ensemble(data_lorenz, library):
+    x, t = data_lorenz
+    library.fit(x)
+    N = library.n_output_features_
+    library.library_ensemble = True
+    xp = library.transform(x)
+    assert N == xp.shape[1] + 1
+    library.ensemble_indices = [0, 1]
+    xp = library.transform(x)
+    assert N == xp.shape[1] + 2
+
+
+@pytest.mark.parametrize(
+    "library",
+    [
+        IdentityLibrary,
+        PolynomialLibrary,
+        FourierLibrary,
+    ],
+)
+def test_bad_library_ensemble(data_lorenz, library):
+    x, t = data_lorenz
+    with pytest.raises(ValueError):
+        library = library(ensemble_indices=-1)
+
+
+# Try various size spatiotemporal inputs for spatiotemporal library
+def test_spatiotemporal_shapes(data_lorenz):
+    data, t = data_lorenz
+    x = np.linspace(0, 10, 10)
+    y = np.linspace(0, 10, 50)
+    X, Y = np.meshgrid(x, y, indexing="ij")
+    library_functions = [lambda x: x, lambda x: x ** 2]
+    library = SpatiotemporalLibrary(
+        library_functions=library_functions, spatiotemporal_variables=[X, Y]
+    )
+    library.fit_transform(data)
+    check_is_fitted(library)
+    x = np.linspace(0, 10, 5)
+    y = np.linspace(0, 10, 10)
+    z = np.linspace(0, 10, 10)
+    X, Y, Z = np.meshgrid(x, y, z, indexing="ij")
+    library = SpatiotemporalLibrary(
+        library_functions=library_functions, spatiotemporal_variables=[X, Y, Z]
+    )
+    library.fit_transform(data)
+    check_is_fitted(library)
+    x = np.linspace(0, 10, 5)
+    y = np.linspace(0, 10, 5)
+    z = np.linspace(0, 10, 5)
+    t = np.linspace(0, 10, 4)
+    X, Y, Z, T = np.meshgrid(x, y, z, t, indexing="ij")
+    library = SpatiotemporalLibrary(
+        library_functions=library_functions, spatiotemporal_variables=[X, Y, Z, T]
+    )
+    library.fit_transform(data)
+    check_is_fitted(library)
+
+
+def test_1D_pdes():
```

Comment:
agreed, yes I'll put some tests for easy PDEs where the solutions can be generated by simple FFTs. I opted against this originally because the PDE-FIND examples all involve loading .mat files that slow down the tests considerably.